activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -1172,23 +1172,26 @@ def where_clause_factory
       end
       alias having_clause_factory where_clause_factory
 
+      DEFAULT_VALUES = {
+        create_with: FROZEN_EMPTY_HASH,
+        readonly: false,
+        where: Relation::WhereClause.empty,
+        having: Relation::WhereClause.empty,
+        from: Relation::FromClause.empty
+      }
+
+      Relation::MULTI_VALUE_METHODS.each do |value|
+        DEFAULT_VALUES[value] ||= FROZEN_EMPTY_ARRAY
+      end
+
+      Relation::SINGLE_VALUE_METHODS.each do |value|
+        DEFAULT_VALUES[value] = nil if DEFAULT_VALUES[value].nil?
+      end
+
       def default_value_for(name)
-        case name
-        when :create_with
-          FROZEN_EMPTY_HASH
-        when :readonly
-          false
-        when :where, :having
-          Relation::WhereClause.empty
-        when :from
-          Relation::FromClause.empty
-        when *Relation::MULTI_VALUE_METHODS
-          FROZEN_EMPTY_ARRAY
-        when *Relation::SINGLE_VALUE_METHODS
-          nil
-        else
-          raise ArgumentError, "unknown relation value #{name.inspect}"
-        end
+        DEFAULT_VALUES.fetch(name)
+      rescue KeyError
```

Comment:
Rather than a `rescue`, can we do the raise in a block passed to `fetch`?

Reply:
This is very internal.. do we care enough to produce our own exception, or can we just leave it up to a blockless `fetch`? It'd change the exception raised, but given this is a "something is broken inside Rails" error, not "something in your app is wrong", that doesn't seem important.

If so, should we then inline this private method into its single caller? `default_value_for(name)` doesn't seem much more descriptive than `DEFAULT_VALUES.fetch(name)`.